ICRCs/ICRC-103/ICRC-103.md

```diff
@@ -0,0 +1,122 @@
+| Status |
+|:------:|
+|Draft|
+
+# ICRC-103: Enhanced Allowance Query Mechanism with Pagination
+
+## 1. Introduction
+
+Although calls to the `icrc2_approve` and `icrc2_transfer_from` methods are recorded in the ledger, it is not possible to determine the allowances that are in effect at some point in time, except by traversing the entire ledger.  This standard introduces an endpoint which will return this information thus making the management of allowances feasible.
+
+ICRC-103 is an extension of the ICRC-2 standard.  
+ICRC-103 specifies a way to list outstanding allowances.
+
+
+## 2. Metadata
+
+A ledger that implements ICRC-103 MUST include `record {name = "ICRC-103"; url = "https://github.com/dfinity/ICRC-1/standards/ICRC-103"}` as part of the output of `icrc1_supported_standards`.
+
+The endpoint introduced in this standard operates in two ways.  In the public version any principal can obtain the outstanding allowances of any other principal. In the private version, the allowances returned by the endpoint must have been issued by the caller (i.e. the caller is the principal controlling the source account of an allowance.)
+Which version of the standard is implemented by a ledger is specified through the `icrc103:public_allowances` metadata.
+
+A ledger that implements ICRC-103 MUST return the metadata entry `icrc103:public_allowances` of type `text`. The possible values are "true" if the allowance data is public and "false" otherwise.
+
+The number of allowances that the ledger can return in response to a query is limited by an implementation-specific maximum, specified through the `icrc103:max_take_value` metadata.
+
+A ledger that implements ICRC-103 MUST return the metadata entry `icrc103:max_take_value` of type `nat`, indicating the precise maximum number of allowances the ledger will return in response to a query.
+
+Metadata entries can be retrieved using method `icrc1_metadata` defined in the ICRC-1 standard as well as `icrc103_collection_metadata` defined below.
```

Comment:
`icrc103_collection_metadata` is not needed, both ICRC-1 and ICRC-7 have a single metadata method (`icrc1_metada` and `icrc7_collection_metadata` respectively) that should also be used by extensions. In the past this topic of additional metadata methods came up in the WG, and the decisions was made that one method in the base standard suffices considering that entry keys are prefixed by their standard.

Reply:
I see -- so these are viewed as the base standards. I thought that `icrc7_collection_metadata` only returns metadata that starts with `icrc7:....`. You're saying that it would return metadata pertaining to any other extention of ICRC_7 that the ledger implements. 
Very good, I'll remove that.